diff --git a/sc/qa/unit/functions_array.cxx b/sc/qa/unit/functions_array.cxx
index 2053c44dfad7..7d290c0f37b5 100644
--- a/sc/qa/unit/functions_array.cxx
+++ b/sc/qa/unit/functions_array.cxx
@@ -29,13 +29,11 @@ void ArrayFunctionsTest::testDubiousArrayFormulasFODS()
     // values when built with -ffp-contract enabled (-ffp-contract=on default on Clang 14,
     // -ffp-contract=fast default when building with optimizations on GCC) on at least aarch64
     // and ppc64le. Thus limit the check only to platforms with consistent results.
-#if defined X86_64
     OUString aDirectoryURL
         = m_directories.getURLFromSrc(u"/sc/qa/unit/data/functions/array/dubious/fods/");
     recursiveScan(test::pass, "OpenDocument Spreadsheet Flat XML", aDirectoryURL,
             "com.sun.star.comp.filter.OdfFlatXml,,com.sun.star.comp.Calc.XMLOasisImporter,com.sun.star.comp.Calc.XMLOasisExporter,,,true",
             FODS_FORMAT_TYPE, SotClipboardFormatId::NONE, 0, false);
-#endif
 }
 
 ArrayFunctionsTest::ArrayFunctionsTest():
