diff --git a/configure.ac b/configure.ac
index 186c472b7d45..5c75fba22fd3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -13559,7 +13609,7 @@ dnl ===================================================================
 dnl Test whether to build libtiff or rely on the system version
 dnl ===================================================================
 
-libo_CHECK_SYSTEM_MODULE([libtiff],[LIBTIFF],[libtiff])
+libo_CHECK_SYSTEM_MODULE([libtiff],[LIBTIFF],[libtiff-4])
 
 dnl ===================================================================
 dnl Test whether to build libwebp or rely on the system version
